test/e2e_flags.go

```diff
@@ -68,7 +70,13 @@ func SetupLoggingFlags() {
 
 // ImagePath is a helper function to transform an image name into an image reference that can be pulled.
 func ImagePath(name string) string {
-	tpl, err := template.New("image").Parse(Flags.ImageTemplate)
+	tpl, err := template.New("image").Funcs(template.FuncMap{
+		"toEnvKey": func(in string) string {
+			underscores := strings.ReplaceAll(in, "-", "_")
```

Comment:
just to ask the question: do we actually need `toEnvKey`/`fromEnv`? Another approach would be for the template to be (though I guess that ends up a pretty ugly template string that you have to build somehow so 🤷):

~~~~
{{with .Name }}
{{if eq . "readiness"}}docker.io/flibble/blobble{{end}}
{{if eq . "varlog"}}docker.io/flibble/bloop{{end}}
..
~~~~

Im wondering a bit if it's worth merging toEnvKey and fromEnv into something like `fromPrefixedEnv("KNATIVE_SERVING_TEST_", name)` since I find it a bit hard to follow the flow with multiple template functions calling each other